File: src/Ecologist.java
/**
 * Клас Еколог
 */
public class Ecologist extends Person implements ServicePlanner {
    private String zone;

    public Ecologist(String name, String zone) {
        super(name, "Еколог");
        this.zone = zone;
    }

    public boolean checkExcess(Measure measure) {
        System.out.println("Еколог " + getName() + " перевіряє перевищення норм...");
        return measure.getValue() > 100; // імітація перевірки
    }

    @Override
    public void planService() {
        System.out.println("Еколог " + getName() + " планує сервісне обслуговування...");
    }

    @Override
    public boolean checkStatus() {
        System.out.println("Еколог " + getName() + " перевіряє статус...");
        return true;
    }
}

File: src/Ecologist.py
# Клас Еколог
class Ecologist(Person, ServicePlanner):
    def __init__(self, name: str, zone: str):
        super().__init__(name, "Еколог")
        self.zone = zone

    def check_excess(self, measure):
        print(f"Еколог {self.get_name()} перевіряє перевищення норм...")
        # імітація перевірки
        return measure.get_value() > 100

    def plan_service(self):
        print(f"Еколог {self.get_name()} планує сервісне обслуговування...")

    def check_status(self) -> bool:
        print(f"Еколог {self.get_name()} перевіряє статус...")
        return True
